pkg/splunk/client/enterprise.go

```diff
@@ -611,7 +613,291 @@ func (c *SplunkClient) DecommissionIndexerClusterPeer(enforceCounts bool) error
+		} `json:"entry"`
+	}{}
+	path := "/services/server/info/server-info"
+	err := c.Get(path, &apiResponseServerRoles)
+	if err != nil {
+		return nil, err
+	}
+	if len(apiResponseServerRoles.Entry) < 1 {
+		return nil, err
+	}
+	return &apiResponseServerRoles.Entry[0].Content, nil
+}
+
+//UpdateDMCGroups dmc* groups with new members
+func (c *SplunkClient) UpdateDMCGroups(dmcGroupName string, groupMembers string) error {
+	endpoint := fmt.Sprintf("%s/services/search/distributed/groups/%s/edit", c.ManagementURI, dmcGroupName)
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(groupMembers))
+	err = c.Do(request, "200, 201, 409", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//UpdateDMCClusteringLabelGroup update respective clustering group
+func (c *SplunkClient) UpdateDMCClusteringLabelGroup(groupName string, groupMembers string) error {
+	endpoint := fmt.Sprintf("%s/services/search/distributed/groups/dmc_indexerclustergroup_%s/edit", c.ManagementURI, groupName)
+	reqBodyClusterGroup := groupMembers + "&default=false"
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(reqBodyClusterGroup))
+	err = c.Do(request, "200, 201, 409", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+//MCAssetBuildTable is the struct for information about asset table
+type MCAssetBuildTable struct {
+	DispatchAutoCancel string `json:"dispatch.auto_cancel"`
+	DispatchBuckets    int64  `json:"dispatch.buckets"`
+}
+
+//GetMonitoringconsoleAssetTable to GET monitoring console asset table data.
+func (c *SplunkClient) GetMonitoringconsoleAssetTable() (*MCAssetBuildTable, error) {
+	apiResponseMCAssetTableBuild := struct {
+		Entry []struct {
+			Content MCAssetBuildTable `json:"content"`
+		} `json:"entry"`
+	}{}
+	path := "/servicesNS/nobody/splunk_monitoring_console/saved/searches/DMC%20Asset%20-%20Build%20Full"
+	err := c.Get(path, &apiResponseMCAssetTableBuild)
+	if err != nil {
+		return nil, err
+	}
+	if len(apiResponseMCAssetTableBuild.Entry) < 1 {
+		return nil, err
+	}
+	return &apiResponseMCAssetTableBuild.Entry[0].Content, nil
+}
+
+//PostMonitoringConsoleAssetTable to build monitoring console asset table. Kicks off the search [Build Asset Table full]
+func (c *SplunkClient) PostMonitoringConsoleAssetTable(apiResponseMCAssetTableBuild *MCAssetBuildTable) error {
+	reqBodyAssetTable := "&trigger_actions=true&dispatch.auto_cancel=" + apiResponseMCAssetTableBuild.DispatchAutoCancel + "&dispatch.buckets=" + strconv.FormatInt(apiResponseMCAssetTableBuild.DispatchBuckets, 10) + "&dispatch.enablePreview=true"
+	endpoint := fmt.Sprintf("%s", c.ManagementURI) + "/servicesNS/nobody/splunk_monitoring_console/saved/searches/DMC%20Asset%20-%20Build%20Full/dispatch"
+	request, err := http.NewRequest("POST", endpoint, strings.NewReader(reqBodyAssetTable))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	err = c.Do(request, "201", nil)
+	if err != nil {
+		return err
+	}
+	return err
+}
```

Comment:
Just ` return err`

Reply:
In either case, we are returning error, so, like 833 to 836 can be replaced with "return err"

Reply:
Oh yes, makes sense. Fixing this in the latest commit